### DISCUSSION TITLE
Difference in the behavior of a form and the isDirty state since v7.39.5

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#9719] [Q&A] [answered]
In my code I noticed a strange behavior that occurs since react-hook-form version 7.39.5.
Specifically, it seems that all forms are dirty by default when they are inside an overlay component, such as a panel or dialog.
For this purpose I have created the following CodeSandbox: [CodeSandbox](https://codesandbox.io/s/great-https-vmsvul?file=/src/App.js).

If you change the version to v7.39.4 and open the dialog you will see that the Save button is disabled because no item is selected in the dropdown, which is required.

But if you change the version to v.7.41.5, which is the latest version, you´ll see that the Save button is enabled; which means that the form somehow is dirty.

I´ve also checked the defaultValues and dirtyFields objects which are available after destructuring them from useForm.
My default values ar an empty object, same as the dirty fields, so since they´re equal isDirty should not be true, but it gets set to true.

As far as i can see, v7.39.5 is the version that introduces this behaviour.

**Comment by leapful** [comment, marked as answer]
@manuelpoelzl 

Since 7.39.5, react-hook-form will strictly check value for dirty state. In your example, there is no default value for dropdown field so its value will be undefined but default value of dropdown component itself is always an empty therefore dirty check is true in this case.

Just add default value as empty string to solve the issue.

Working demo: https://codesandbox.io/s/xenodochial-snowflake-cscej9?file=/src/App.js

**Reply by manuelpoelzl** [reply]
Thanks for the demo!
Is this a wanted behavior that this does not work on the controller level?
For example, if I specify an empty string as DefaultValue for the controller, I still have the behavior that isDirty is true.
Sample: [CodeSandbox Controller defaultValue](https://codesandbox.io/s/suspicious-firefly-v39v29?file=/src/App.js)

**Reply by leapful** [reply]
@manuelpoelzl 

Yes, it's the expected behavior, default value at field level on `Controller` component is not for dirty checking of the whole form state.

**Reply by manuelpoelzl** [reply]
Is there any documentation that describes the differences between the defaultValues for useForm and the property on the controller?

If I understand this correctly, the defaultValue property on the controller can then be removed, since isDirty is determined based on the parameter passed to useForm.

Do I understand this correctly, or am I on a wrong track?

**Reply by leapful** [reply]
@manuelpoelzl 

The documentation `formState.isDirty` is described here: https://react-hook-form.com/api/useform/formstate that already mentioned `defaultValues` of `useForm` hook is single source of truth for dirty checking.

Property `defaultValue` of `Controller` component is more related to behavior of `defaultValue` in real HTML field element which is not related to dirty checking.

**Reply by manuelpoelzl** [reply]
Now I got it !
Thank you very much for your effort and the explanations!
